test/regress_dns.c

```diff
@@ -1203,19 +1206,56 @@ dns_nameservers_no_default_test(void *arg)
 	 * EVDNS_BASE_INITIALIZE_NAMESERVERS|EVDNS_BASE_NAMESERVERS_NO_DEFAULT
 	 * because we cannot mock "/etc/resolv.conf" (yet). */
 
-	evdns_base_resolv_conf_parse(dns,
+	ok = evdns_base_resolv_conf_parse(dns,
 		DNS_OPTIONS_ALL|DNS_OPTION_NAMESERVERS_NO_DEFAULT, RESOLV_FILE);
+	tt_int_op(ok, ==, EVDNS_ERROR_FAILED_TO_OPEN_FILE);
 	tt_int_op(evdns_base_get_nameserver_addr(dns, 0, NULL, 0), ==, -1);
 	tt_int_op(evdns_base_get_nameserver_fd(dns, 0), ==, -1);
 
-	evdns_base_resolv_conf_parse(dns, DNS_OPTIONS_ALL, RESOLV_FILE);
+	ok = evdns_base_resolv_conf_parse(dns, DNS_OPTIONS_ALL, RESOLV_FILE);
+	tt_int_op(ok, ==, EVDNS_ERROR_FAILED_TO_OPEN_FILE);
 	tt_int_op(evdns_base_get_nameserver_addr(dns, 0, NULL, 0), ==, sizeof(struct sockaddr));
 	tt_int_op(evdns_base_get_nameserver_fd(dns, 0), !=, -1);
 
 end:
 	if (dns)
 		evdns_base_free(dns, 0);
 }
+
+static void
+dns_nameservers_no_nameservers_configured_test(void *arg)
+{
+	struct basic_test_data *data = arg;
+	struct event_base *base = data->base;
+	struct evdns_base *dns = NULL;
+	int fd = -1;
+	char *tmpfilename = NULL;
+	const char filecontents[] = "# tmp empty resolv.conf\n";
+	const size_t filecontentssize = sizeof(filecontents);
+	int ok;
+	
+	fd = regress_make_tmpfile(filecontents, filecontentssize, &tmpfilename);
+	if (fd < 0)
+	{
+		tt_skip();
+	}
+
+	dns = evdns_base_new(base, 0);
+	tt_assert(dns);
+
+	ok = evdns_base_resolv_conf_parse(dns, DNS_OPTIONS_ALL, tmpfilename);
+	tt_int_op(ok, ==, EVDNS_ERROR_NO_NAMESERVERS_CONFIGURED);
+
+end:
```

Comment:
Can you also free `dns` here to fix [ASAN](https://github.com/libevent/libevent/actions/runs/4010248966/jobs/6886553273):

```
   if (dns)
       evdns_base_free(dns, 0);
```

```
Indirect leak of 1176 byte(s) in 17 object(s) allocated from:
    #0 0x560cc5beb24e in malloc (/home/runner/work/libevent/libevent/build/bin/regress+0x12a24e) (BuildId: a5ab88d462ea2fd0062b7da045b035b0ec100f65)
    #1 0x7fe0ec5628d5 in event_mm_calloc_ /home/runner/work/libevent/libevent/event.c:3568:7
    #2 0x7fe0ec671a40 in evdns_base_parse_hosts_line /home/runner/work/libevent/libevent/evdns.c:5086:8
    #3 0x7fe0ec655f98 in evdns_base_load_hosts_impl /home/runner/work/libevent/libevent/evdns.c:5131:4
    #4 0x7fe0ec655af9 in evdns_base_load_hosts /home/runner/work/libevent/libevent/evdns.c:5150:8
    #5 0x7fe0ec65286c in evdns_base_resolv_conf_parse_impl /home/runner/work/libevent/libevent/evdns.c:4531:3
    #6 0x7fe0ec652634 in evdns_base_resolv_conf_parse /home/runner/work/libevent/libevent/evdns.c:4485:8
    #7 0x560cc5cb61aa in dns_nameservers_no_nameservers_configured_test /home/runner/work/libevent/libevent/test/regress_dns.c:1246:7
    #8 0x560cc5d6824c in testcase_run_bare_ /home/runner/work/libevent/libevent/test/tinytest.c:175:4
    #9 0x560cc5d67b57 in testcase_run_forked_ /home/runner/work/libevent/libevent/test/tinytest.c:272:12
    #10 0x560cc5d677d3 in testcase_run_one /home/runner/work/libevent/libevent/test/tinytest.c:339:13
    #11 0x560cc5d69389 in tinytest_main /home/runner/work/libevent/libevent/test/tinytest.c:560:20
    #12 0x560cc5d320b1 in main /home/runner/work/libevent/libevent/test/regress_main.c:541:6
    #13 0x7fe0ec201d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f) (BuildId: 69389d485a9793dbe873f0ea2c93e02efaa9aa3d)
```

Reply:
No worries, I will do this by myself